### PR TITLE
[INLONG-10607][SDK] Transform SQL support arithmetic functions(Including log10, log2, log and exp)

### LINKED ISSUE #10607: [Feature][SDK] Transform SQL support arithmetic functions(Including log10, log2, log and exp)
### Description

Enable transform SQL to support the following FlinkSQL arithmetic functions:
- log10(numeric): returns the base 10 logarithm of numeric
- log2(numeric): returns the base 2 logarithm of numeric
- log(numeric) or log(numeric1, numeric2): When called with one argument, returns the natural logarithm of numeric. When called with two arguments, this function returns the logarithm of numeric2 to the base numeric1.
- exp(numeric): returns e raised to the power of numeric

Parent issue:
https://github.com/apache/inlong/issues/10022

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yfsn666**
Fixes #10607 

### Motivation

Transform SQL support arithmetic functions(Including log10, log2, log and exp)

### Modifications

Add four arithmetic function classes: Log2Function, Log10Function, LogFunction, and ExpFunction. Also, add the corresponding unit test codes

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests in `TestTransformArithmeticFunctionsProcessor.java`

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? not documented

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ExpFunction.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.math.BigDecimal;
+
+/**
+ * ExpFunction
+ * description: exp(numeric)--returns e raised to the power of numeric
+ */
+public class ExpFunction implements ValueParser {
+
+    private ValueParser number;
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ExpFunction.java:34]:
```suggestion
    private ValueParser parse;
```

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/Log10Function.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.math.BigDecimal;
+
+/**
+ * Log10Function
+ * description: log10(numeric)--returns the base 10 logarithm of numeric
+ */
+public class Log10Function implements ValueParser {
+
+    private ValueParser number;
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/Log10Function.java:34]: ditto

Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/Log10Function.java:34]:
```suggestion
    private ValueParser parse;
```

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/Log2Function.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.math.BigDecimal;
+
+/**
+ * Log2Function
+ * description: log2(numeric)--returns the base 2 logarithm of numeric
+ */
+public class Log2Function implements ValueParser {
+
+    private ValueParser number;
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/Log2Function.java:34]: parse

Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/Log2Function.java:34]:
```suggestion
    private ValueParser parser;
```

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/LogFunction.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+/**
+ * LogFunction
+ * description: log(numeric) or log(numeric1, numeric2)--When called with one argument, returns the natural logarithm
+ * of numeric. When called with two arguments, this function returns the logarithm of numeric2 to the base numeric1
+ */
+public class LogFunction implements ValueParser {
+
+    private ValueParser base;
+    private ValueParser number;
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/LogFunction.java:38]:
```suggestion
    private ValueParser baseParse;
    private ValueParser parse;
```

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java [hunk 5]
```diff
@@ -117,4 +117,72 @@ public void testLnFunction() throws Exception {
...
+        String transformSql = "select log10(numeric1) from source";
+        TransformConfig config = new TransformConfig(csvSource, kvSink, transformSql);
+        // case1: log10(1)
+        TransformProcessor processor = new TransformProcessor(config);
+        List<String> output1 = processor.transform("1|4|6|8", new HashMap<>());
+        Assert.assertTrue(output1.size() == 1);
+        Assert.assertEquals(output1.get(0), "result=0.0");
+        // case2: log10(1000)
+        List<String> output2 = processor.transform("1000|4|6|8", new HashMap<>());
+        Assert.assertTrue(output2.size() == 1);
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java:132]:
```suggestion
         Assert.assertEquals(1, output1.size());
        Assert.assertEquals(output1.get(0), "result=0.0");
        // case2: log10(1000)
        List<String> output2 = processor.transform("1000|4|6|8", new HashMap<>());
        Assert.assertEquals(1, output2.size());
```

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java [hunk 6]
```diff
@@ -117,4 +117,72 @@ public void testLnFunction() throws Exception {
...
+    }
+
+    @Test
+    public void testLogFunction() throws Exception {
+        String transformSql1 = "select log(numeric1) from source";
+        TransformConfig config1 = new TransformConfig(csvSource, kvSink, transformSql1);
+        // case1: ln(1)
+        TransformProcessor processor1 = new TransformProcessor(config1);
+        List<String> output1 = processor1.transform("1|4|6|8", new HashMap<>());
+        Assert.assertTrue(output1.size() == 1);
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java:158]:
```suggestion
        Assert.assertEquals(1, output1.size());
```

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java [hunk 7]
```diff
@@ -117,4 +117,72 @@ public void testLnFunction() throws Exception {
...
+        Assert.assertEquals(output1.get(0), "result=0.0");
+        String transformSql2 = "select log(numeric1, numeric2) from source";
+        TransformConfig config2 = new TransformConfig(csvSource, kvSink, transformSql2);
+        // case2: log2(8)
+        TransformProcessor processor2 = new TransformProcessor(config2);
+        List<String> output2 = processor2.transform("2|8|6|8", new HashMap<>());
+        Assert.assertTrue(output2.size() == 1);
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java:165]:
```suggestion
        Assert.assertEquals(1, output2.size());
```

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java [hunk 8]
```diff
@@ -117,4 +117,72 @@ public void testLnFunction() throws Exception {
...
+        Assert.assertEquals(output2.get(0), "result=3.0");
+        // case3: log10(100)
+        TransformProcessor processor3 = new TransformProcessor(config2);
+        List<String> output3 = processor3.transform("10|100|6|8", new HashMap<>());
+        Assert.assertTrue(output3.size() == 1);
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java:170]:
```suggestion
        Assert.assertEquals(1, output3.size());
```

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java [hunk 9]
```diff
@@ -117,4 +117,72 @@ public void testLnFunction() throws Exception {
...
+        String transformSql = "select exp(numeric1) from source";
+        TransformConfig config = new TransformConfig(csvSource, kvSink, transformSql);
+        // case1: e^0
+        TransformProcessor processor = new TransformProcessor(config);
+        List<String> output1 = processor.transform("0|4|6|8", new HashMap<>());
+        Assert.assertTrue(output1.size() == 1);
+        Assert.assertEquals(output1.get(0), "result=1.0");
+        // case2: e^2
+        List<String> output2 = processor.transform("2|4|6|8", new HashMap<>());
+        Assert.assertTrue(output2.size() == 1);
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java:185]:
```suggestion
        Assert.assertEquals(1, output1.size());
        Assert.assertEquals(output1.get(0), "result=1.0");
        // case2: e^2
        List<String> output2 = processor.transform("2|4|6|8", new HashMap<>());
        Assert.assertEquals(1, output2.size());
```

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ExpFunction.java (hunk 1 above)
Comment (yfsn666) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ExpFunction.java]: renamed the relevant variables

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java (hunk 5 above)
Comment (yfsn666) [reply inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java]: optimized the test code

**Review by aloyszhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

**Pull request merged**